sdk/lib/cmlib/hiveinit.c

```diff
@@ -44,6 +45,129 @@ HvpVerifyHiveHeader(
     return TRUE;
 }
 
+BOOLEAN CMAPI
+HvRestoreFromLog(
+    _In_ PHHIVE Hive,
+    _Inout_ PHBASE_BLOCK BaseBlock
+)
+{
+    HBASE_BLOCK logBaseBlock;
+    ULONG offset = 0;
+    ULONG indexInLog;
+    BOOLEAN isSuccess;
+    UCHAR dirtyVectBuffer[HSECTOR_SIZE];
+    UCHAR buffer[HBLOCK_SIZE];
+
+    ASSERT(sizeof(HBASE_BLOCK) >= (HSECTOR_SIZE * Hive->Cluster));
+
+    // Read log file header
+    isSuccess = Hive->FileRead(Hive,
+                            HFILE_TYPE_LOG,
+                            &offset,
+                            &logBaseBlock,
+                            Hive->Cluster * HSECTOR_SIZE);
+
+    if (!isSuccess)
+    {
+        DPRINT1("Read LOG file failed\n");
+        return FALSE;
+    }
+
+    // Validate log header
+    if (!HvpVerifyHiveHeader(&logBaseBlock, HFILE_TYPE_LOG))
+    {
+        DPRINT1("LOG header corrupted\n");
+        return FALSE;
+    }
```

Comment:
What has been done until now, is for restoring the header when handling the `RecoverHeader` error case during loading.
What is following next below, is for handling the `RecoverData` error case during loading.
It would be nice to have these in two separate recovery functions.